components/scream/src/physics/shoc/shoc_functions_f90.hpp

```diff
@@ -16,9 +17,54 @@
 namespace scream {
 namespace shoc {
 
-///////////////////////////////////////////////////////////////////////////////
 
+///////////////////////////////////////////////////////////////////////////////
 // Converted subroutine helpers go here.
+struct SHOCGridData{
+static constexpr size_t NUM_ARRAYS = 4; 
+static constexpr size_t NUM_ARRAYS_i = 2; 
+
+    
+ // Inputs 
+ Int shcol, nlev, nlevi;
+ Real *zt_grid, *zi_grid, *pdel;
+
+ // In/out
+ Real *dz_zt, *dz_zi, *rho_zt;
+
+
+ SHOCGridData(Int shcol_, Int nevl_, Int nlevi_);
+ SHOCGridData(const SHOCGridData& rhs);
+ SHOCGridData& operator=(const SHOCGridData& rhs);
```

Comment:
Could you explain what the 3 lines above do? Are they ways of initializing the SHOCGridData class? I haven't seen such a thing within a struct before...

Reply:
Nevermind. I got it.